src/events/autoMod.js

```diff
@@ -0,0 +1,24 @@
+const fs = require("fs");
+
+class CacheManager {
+     constructor(val) {
+          this.cache = String(val).split(/[\n\r]/);
+          this.cache = this.cache.filter((url) => url.length > 0);
+          this.cache = new Map(this.cache.map((url) => [url, true]));
+     }
+     has(val) {
+          return this.cache.has(val);
+     }
+}
+
+const cache = new CacheManager(fs.readFileSync("data/scam-urls.txt"));
+
+module.exports = {
+     name: "messageCreate",
+     once: false,
+     execute(msg) {
+          if (cache.has(msg.content)) {
+               msg.delete();
```

Comment:
maybe return this? then it could theoretically be awaited outside of this func

Reply:
There is no reason for awaiting Message.prototype.delete method, but I noted that one!

Reply:
IMO since this function is essentially an async function it should return a  promise. No matter if we want to actually wait for it, it's just more transparent and intuitive that way 🤷🏻‍♂️